src/log_outgoing_requests/formatters.py

```diff
@@ -0,0 +1,31 @@
+import logging
+import textwrap
+
+
+class HttpFormatter(logging.Formatter):
+    def _formatHeaders(self, d):
+        return "\n".join(f"{k}: {v}" for k, v in d.items())
+
+    def formatMessage(self, record):
+        result = super().formatMessage(record)
+        if record.name == "requests":
+            result += textwrap.dedent(
+                """
+                ---------------- request ----------------
+                {req.method} {req.url}
+                {reqhdrs}
+
+                {req.body}
+                ---------------- response ----------------
+                {res.status_code} {res.reason} {res.url}
+                {reshdrs}
+                
+            """
+            ).format(
+                req=record.req,
+                res=record.res,
+                reqhdrs=self._formatHeaders(record.req.headers),
+                reshdrs=self._formatHeaders(record.res.headers),
```

Comment:
I like the header information for debugging, but I'm not sure if we can store that in logs. @alextreme can pitch in on that.